### ISSUE TITLE
Unnecessary Poetry install build step 

### ISSUE THREAD

**Issue by moranimu** [terraform-aws-modules/terraform-aws-lambda#436]
## Description

Since version 4.3.0 we encountered that when trying to package Poetry dependencies, a new build step is added to the build plan - poetry install, even though we did not set poetry_install = true.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version: 4.3.0 and further
Terraform v1.0.8
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v4.58.0
+ provider registry.terraform.io/hashicorp/external v2.3.1
+ provider registry.terraform.io/hashicorp/local v2.4.0
+ provider registry.terraform.io/hashicorp/null v3.2.1

## Reproduction Code [Required]

1. Create a Poetry project in your module with .toml and .lock 
2. run terraform apply

## Expected behavior

a build_plan that includes only pip install

## Actual behavior

A build plan includes 2 steps: pip install (required) and poverty install (not required)

### Terminal Output Screenshot(s)

build_plan = [
                  [
                      "pip",
                      "python3.8",
                      "/builds/lambda/requirements.txt",
                      ,
                      ,
                  ],
                  [
                      "poetry",
                      "python3.8",
                      "/builds/lambda",
                      ,
                  ],
                  [
                      "zip",
                      "/builds/lambda",
                      ,
                    ],
                ]

**Comment by moranimu** [comment]
Any update on this issue?

**Comment by antonbabenko** [comment]
@pdecat What do you think about this issue?

**Comment by pdecat** [comment]
Went under my radar, I'll look into it.

**Comment by pdecat** [comment]
Hi @moranimu,

> Since version 4.3.0 we encountered that when trying to package Poetry dependencies, a new build step is added to the build plan - poetry install, even though we did not set poetry_install = true.

`poetry_install` is only taken into account if the `source` parameter is [passed as a dict](https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/v4.14.0/package.py#L793).
When source is passed as a string pointing to a directory, [discovery of the build system is automatically performed](https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/v4.14.0/package.py#L775).
If both a `requirements.txt` file and a `pyproject.toml` file defining poetry as the build system are present, both dependencies managers are executed.
Maybe this should be changed to only process the first dependencies manager found. But triggering both maybe actually be desired in some cases.

In the meantime, you should be able to restore previous behavior by converting the `source` field from a string into a dict, e.g.:

Before:
```
  source_path = "${path.module}/../fixtures/python3.9-app-poetry"
```

After:
```
  source_path = [
    {
      path           = "${path.module}/../fixtures/python3.9-app-poetry"
    }
  ]
```

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.16.0 :tada:
